Project/DataAccess/Context/AppContext.cs

```diff
@@ -79,5 +80,22 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
 
             base.OnModelCreating(modelBuilder);
         }
+
+        /// <summary>
+        /// Saves all changes made in this context to the underlying database.
+        /// </summary>
+        /// <returns>
+        /// Amount of transaction that has been confirmed.
+        /// </returns>
+        public override int SaveChanges()
+        {
+            PhotoLike.Local.Where(ph => ph.Photo == null && ph.User == null).ToList().ForEach(ph => PhotoLike.Remove(ph));
+
+            Comments.Local.Where(c => c.Photo == null && c.User == null).ToList().ForEach(c => Comments.Remove(c));
+
+            CommentLike.Where(cl => cl.Comment == null && cl.User == null).ToList().ForEach(cl => CommentLike.Remove(cl));

```

Comment:
What **'Local'** do? And why you dont have it on comment like?

Reply:
Yes, thanks, it's my mistake. Local represents changes that are not yet saved to database.

Reply:
Ok, I move it to typos

Reply:
Also, I do not get why you need to convert them to list? if it only to **ForEach** method seems not to worth it. But dont bother on this now. I will open an issue